pkg/provision/providers/vm/disk.go

```diff
@@ -49,15 +50,20 @@ func (p *Provisioner) CreateDisks(state *State, nodeReq provision.NodeRequest) (
 
 		defer diskF.Close() //nolint:errcheck
 
-		err = diskF.Truncate(int64(disk.Size))
+		if err = diskF.Truncate(int64(disk.Size)); err != nil {
+			return nil, err
+		}
+
+		if err = syscall.Fallocate(int(diskF.Fd()), 0, 0, int64(disk.Size)); err != nil {
+			fmt.Fprintf(os.Stderr, "WARNING: failed to allocate disk space for %q (size %d): %s", diskPath, disk.Size, err)
+		}
+
 		diskPaths[i] = diskPath
 	}
 
 	if len(diskPaths) == 0 {
-		err = fmt.Errorf("node request must have at least one disk defined to be used as primary disk")
-
-		return
+		return nil, fmt.Errorf("node request must have at least one disk defined to be used as primary disk")
```

Comment:
Maybe we should just drop named returns here? It looks kinda inconsistent  with whats written above in this function.

Reply:
I see named returns as documentation here, so I kept them, but all `return`s are explicit, as otherwise function returning `[]string, error` is confusing

Reply:
Oh, okay!